eng/scripts/Language-Settings.ps1

```diff
@@ -5,6 +5,50 @@ $packagePattern = "*.zip"
 $MetadataUri = "https://raw.githubusercontent.com/Azure/azure-sdk/master/_data/releases/latest/python-packages.csv"
 $BlobStorageUrl = "https://azuresdkdocs.blob.core.windows.net/%24web?restype=container&comp=list&prefix=python%2F&delimiter=%2F"
 
+function Get-AllPackageInfoFromRepo
+{
+  $allPackageProps = @()
+  Push-Location $RepoRoot
+  $allSetupProps = $null
+  try{
+    pip install packaging==20.4 -q -I
+    $allSetupProps = (python -c "import sys; import glob; import os; sys.path.append(os.path.join('scripts', 'devops_tasks')); from common_tasks import get_package_properties; \
+    obj=[get_package_properties(os.path.dirname(p)) for p in glob.glob('sdk/*/*/setup.py', recursive=True) \
+    if os.path.basename(os.path.dirname(p)) != 'azure-mgmt' and os.path.basename(os.path.dirname(p)) != 'azure' and os.path.basename(os.path.dirname(p)) != 'azure-storage']; \
+    print(obj)")
+  }
+  catch
+  {
+    # This is soft error and failure is expected for python metapackages
+    LogError "Failed to get all package properties"
+  }
+  Pop-Location
+  $allMatches = $allSetupProps | Select-String '\(.*?\)' -AllMatches
+  foreach ($matchInfo in $allMatches.Matches)
+  {
+    $matchValue = $matchInfo.Value -Split " "
```

Comment:
We should see if we can share this method with the Get-python-PackageInfoFromRepo method in some way.

Reply:
I planned to call `Get-AllPackageInfoFromRepo` directly from `Package-Properties` only if it is available. Which should only be the case for python.

Reply:
So is your plan to delete Get-python-PackageInfoFromRepo then? If you plan to keep both then I think we can implement them in shared way to not duplicate this work.

Reply:
Yes I plan to get rid of `Get-python-PackageInfoFromRepo` since there will not be calls to it anymore.

Reply:
OK If that is the plan then I'm fine not refactoring these but lets make sure we have a tracking issue to clean these up as I don't want the duplication to stay around.